### ISSUE TITLE
Opaque structs are not pretty printed correctly

### ISSUE THREAD

**Issue by TomMD** [GaloisInc/llvm-pretty#16]
At least for LLVM 3.8 it appears a valid pretty printing of an opaque struct is:

```
%struct.foo = type { %struct.bar *, i32}
%struct.bar = type opaque
```

However, llvm-pretty is currently producing:

```
%struct.foo = type { opaque*, i32 }
```

**Comment by elliottt** [comment]
Is this the result of parsing with `llvm-pretty-bc-parser`? If so, it might be that they just aren't being parsed properly.

**Comment by TomMD** [comment]
Yes.  A full test is:

```
tommd@HalfAndHalf /private/tmp% cat llvm.c
typedef struct bar_st bar;
struct foo_st {
    bar *bar_val;
    int int_val;
};

void test(struct foo_st foo)
{
}
tommd@HalfAndHalf /private/tmp% clang -c -emit-llvm llvm.c
```

The llvm-pretty{,-bc-parser} output:

```
tommd@HalfAndHalf /private/tmp% llvm-disasm llvm.bc
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

; llvm.bc
target datalayout = "e-m:o-i64:64-f80:128-n8:16:32:64-S128"
%struct.foo_st = type { opaque*, i32 }
```

snip.

The llvm-dis output:

```
tommd@HalfAndHalf /private/tmp% llvm-dis llvm.bc && cat llvm.ll
; ModuleID = 'llvm.bc'
target datalayout = "e-m:o-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-apple-macosx10.12.0"

%struct.bar_st = type opaque
%struct.foo_st = type { %struct.bar_st*, i32 }
```

snip.

**Comment by TomMD** [comment]
Expanding on "yes" that parse is dropping too much information to reliably reconstruct things.  The matching AST is:

```
modTypes =  [TypeDecl{typeName = Ident "struct.foo_st",
                         typeValue = Struct [PtrTo Opaque, PrimType (Integer 32)]}
                      ],
```

**Comment by TomMD** [comment]
This was fixed by GaloisInc/llvm-pretty-bc-parser#55
